reference-implementation/lib/writable-stream.js

```diff
@@ -163,13 +163,37 @@ function WritableStreamAbort(stream, reason) {
   return promise;
 }
 
-function WritableStreamFinishAbort(stream) {
-  const error = new TypeError('Aborted');
-
+function WritableStreamError(stream, error) {
   stream._state = 'errored';
   stream._storedError = error;
 
-  WritableStreamRejectPromisesInReactionToError(stream);
+  // TODO: layering violation?
+  ResetQueue(stream._writableStreamController);
```

Comment:
@tyoshino the stream is not supposed to reach into its controller like this, right? What do you suggest for fixing this?